### ISSUE TITLE
The color picker didn't set to the color selected

### ISSUE THREAD

**Issue by xmeow** [LLK/scratch-blocks#986]
### Expected Behavior

The color picker 'set pen color to ' block in pen catalog didn't reponse to the color selected

### Actual Behavior

It should, but the rounded color picker just keep red. It is interesting that argValues give the correct color value.

### Steps to Reproduce
Take a try at pen catalog from vertical playground.
http://llk.github.io/scratch-blocks/playgrounds/tests/vertical_playground.html

### Operating System and Browser

Windows 10 and chrome 58

**Comment by rschamp** [comment]
@rachel-fenichel Do you think this is related to the bug that affected click targets for other fields?

**Comment by rachel-fenichel** [comment]
If it's storing the correct value but not changing the block, no, it's not the same bug.

**Comment by towerofnix** [comment]
> Looks like this was accidentally closed by the merge/06-28 (the commit that "fixes" this was targeting an issue in google/blockly, not the issue with the same ID in LLK/scratch-blocks.)

**Comment by xmeow** [comment]
It seems like that  `this.isShadow()` in
```
  // Render block fill
  var fillColour = (this.isGlowingBlock_ || this.isShadow()) ?
      this.getColourSecondary() : this.getColour();
```
always return ture and ColourSecondary(#FF0000) is used for color picker background.
The code block locate at core/block_render_svg_vertical.js

**Comment by paulkaplan** [comment]
@xmeow thanks for looking into that! That line was added here https://github.com/LLK/scratch-blocks/pull/945 I guess we overlooked the color block being a shadow that changes color. 

@rachel-fenichel thoughts on the best way to fix this? Does the `FieldColour` need secondary and tertiary colors? We might be able to just change the `setValue` function to set the primary, secondary and tertiary to the new value. That would be changing [these lines](https://github.com/LLK/scratch-blocks/blob/develop/core/field_colour.js#L109-L113) from 
```javascript
    this.sourceBlock_.setColour(
      colour,
      this.sourceBlock_.getColourSecondary(),
      this.sourceBlock_.getColourTertiary()
    );
```
to
```javascript
    this.sourceBlock_.setColour(colour, colour, colour);
```

 That is just the first way that comes to mind, thoughts?
